src/databricks/labs/remorph/snow/lca_utils.py

```diff
@@ -56,12 +57,17 @@ def check_for_unsupported_lca(
 def unalias_lca_in_select(expr: exp.Expression) -> exp.Expression:
     if not isinstance(expr, exp.Select):
         return expr
+
     root_select = build_scope(expr)
+
+    if root_select is None:
+        raise ValueError("Failed to build scope for the select expression")
+
     # We won't search inside nested selects, they will be visited separately
     nested_selects = {*root_select.derived_tables, *root_select.subqueries}
     alias_info = _find_aliases_in_select(expr)
-    where_ast: Expression = expr.args.get("where")
-    if where_ast:
+    where_ast = expr.args.get("where")
+    if where_ast and isinstance(where_ast, Expression):
```

Comment:
Do we really need this checks?

Reply:
Since it could be Expression | None, we are checking for this.